Last-Update: 2013-09-14
Forwarded: not-needed
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
From: Andrew Borodin <aborodin@vmail.ru>
Origin: upstream, https://www.midnight-commander.org/changeset/446748619b5cb77ea63360a81bb16cd2dd2caff7
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=714369
Bug-MC: https://www.midnight-commander.org/ticket/3047
Description: fix mc arguments handling
 Following cases from command line are possible:
  * 'mc' (no arguments):
     active panel uses current directory
     passive panel uses "other_dir" from panels.ini
  * 'mc dir1 dir2' (two arguments):
     active panel uses dir1
     right panel uses dir2
  * 'mc dir1' (single argument):
     active panel uses dir1
     passive panel uses "other_dir" from panels.ini

--- a/doc/man/mc.1.in
+++ b/doc/man/mc.1.in
@@ -171,13 +171,14 @@
 .I \-X, \-\-no\-x11
 Do not use X11 to get the state of modifiers Alt, Ctrl, Shift
 .PP
 If both paths are specified, the first path name is the directory to show
-in the left panel; the second path name is the directory to be shown in
-the right panel.
+in the active panel; the second path name is the directory to be shown in
+the other panel.
 .PP
 If one path is specified, the path name is the directory to show
-in the active panel; current directory is shown in the passive panel.
+in the active panel; value of "other_dir" from panels.ini is the directory
+to be shown in the passive panel.
 .PP
 If no paths are specified, current directory is shown in the active panel;
 value of "other_dir" from panels.ini is the directory to be shown in
 the passive panel.
--- a/doc/man/ru/mc.1.in
+++ b/doc/man/ru/mc.1.in
@@ -148,15 +148,16 @@
 .I \-X, \-\-no\-x11
 Не использовать X11 для получения состояния модификаторов Alt, Ctrl, Shift.
 .PP
 Если в командной строке указаны два каталога, первый будет отображаться
-в левой панели, второй \- в правой.
+в активной панели, второй \- в другой панели.
 .PP
 Если указан только один каталог, он будет отображён в активной панели,
-во второй панели будет отображён текущий каталог.
+во второй панели будет отображён каталог, указанный в параметре "other_dir"
+в файле panels.ini.
 .PP
 Если каталоги не указаны, в активной панели будет отображён текущий каталог,
-а во второй панели \- каталог, указанный в параметре "other_dir" в файла panels.ini.
+а во второй панели \- каталог, указанный в параметре "other_dir" в файле panels.ini.
 .\"NODE "Overview"
 .SH "Главное окно программы"
 Главное окно программы Midnight Commander состоит из трех полей. Два
 поля, называемые "панелями", идентичны по структуре и обычно отображают
--- a/src/filemanager/midnight.c
+++ b/src/filemanager/midnight.c
@@ -583,17 +583,17 @@
     /*
      * Following cases from command line are possible:
      * 'mc' (no arguments):            mc_run_param0 == NULL, mc_run_param1 == NULL
      *                                 active panel uses current directory
-     *                                 passive panel uses "other_dir" from ini
+     *                                 passive panel uses "other_dir" from panels.ini
      *
      * 'mc dir1 dir2' (two arguments): mc_run_param0 != NULL, mc_run_param1 != NULL
-     *                                 left panel uses mc_run_param0
-     *                                 right panel uses mc_run_param1
+     *                                 active panel uses mc_run_param0
+     *                                 passive panel uses mc_run_param1
      *
      * 'mc dir1' (single argument):    mc_run_param0 != NULL, mc_run_param1 == NULL
-     *                                 active panel uses current directory
-     *                                 passive panel uses mc_run_param0
+     *                                 active panel uses mc_run_param0
+     *                                 passive panel uses "other_dir" from panels.ini
      */
 
     /* Set up panel directories */
     if (boot_current_is_left)
@@ -618,10 +618,10 @@
         }
         else                    /* mc_run_param0 != NULL && mc_run_param1 == NULL */
         {
             /* one argument */
-            current_dir = NULL; /* assume current dir */
-            other_dir = (char *) mc_run_param0;
+            current_dir = (char *) mc_run_param0;
+            other_dir = saved_other_dir;        /* from ini */
         }
     }
     else
     {
@@ -639,16 +639,16 @@
         }
         else if (mc_run_param0 != NULL && mc_run_param1 != NULL)
         {
             /* two arguments */
-            current_dir = mc_run_param1;
-            other_dir = (char *) mc_run_param0;
+            current_dir = (char *) mc_run_param0;
+            other_dir = mc_run_param1;
         }
         else                    /* mc_run_param0 != NULL && mc_run_param1 == NULL */
         {
             /* one argument */
-            current_dir = NULL; /* assume current dir */ ;
-            other_dir = (char *) mc_run_param0;
+            current_dir = (char *) mc_run_param0;
+            other_dir = saved_other_dir;        /* from ini */
         }
     }
 
     /* 1. Get current dir */
